spyderlib/config/user.py

```diff
@@ -117,8 +118,27 @@ def _write_file(fname):
                 raise(e)
 
     def filename(self):
+        """Defines the name of the configuration file to use."""
+        self._filename = getattr(self, '_filename', None)
+        self._root_path = getattr(self, '_root_path', None)

```

Comment:
I don't like this. Please add `_filename` and `_root_path` in `__init__`, and set them to `None` with a comment above them saying that they are need for projects.